src/main/java/com/sarality/sync/APISyncUpdateDataFetcher.java

```diff
@@ -0,0 +1,42 @@
+package com.sarality.sync;
+
+import com.sarality.db.Column;
+import com.sarality.db.Table;
+import com.sarality.db.query.Operator;
+import com.sarality.db.query.Query;
+import com.sarality.db.query.SimpleQueryBuilder;
+import com.sarality.db.query.SimpleQueryClause;
+
+import java.util.List;
+
+import hirondelle.date4j.DateTime;
+
+/**
+ * Sync data fetcher for updated records - which have a global Id and were updated after last sync time
+ *
+ * @author Satya@ (Satya Puniani)
+ */
+
+public class APISyncUpdateDataFetcher<T> implements APISyncDataFetcher<T> {
+
+  private final Table<T> table;
+  private final Column globalIdColumn;
+  private final Column timeStampColumn;
+
+  public APISyncUpdateDataFetcher(Table<T> table, Column globalIdColumn, Column timeStampColumn) {
+    this.table = table;
+    this.globalIdColumn = globalIdColumn;
+    this.timeStampColumn = timeStampColumn;
+  }
+
+  
+  @Override
+  public List<T> fetch(DateTime syncDate) {
+    Query newQuery = new SimpleQueryBuilder()
+        .withFilter(new SimpleQueryClause(globalIdColumn, Operator.IS_NOT_NULL))
+        .withFilter(timeStampColumn, Operator.GREATER_THAN_EQUAL_TO, syncDate)
```

Comment:
Do we need to worry about creationTimestamp as well here, because other we'll get new data both here as well as in new data fetcher.

Makes me wonder if we really need just one of them and in the request generator we figure out if we need to generate a Create request or an Update request based on the createdTimestamp relative to last sync timestamp.

Either way, we should make sure we don't get duplicates